test/file/file_close_test.cpp

```diff
@@ -24,178 +24,84 @@
-  EXPECT_NO_THROW(hdf5::file::open("testclose.h5",
-				   hdf5::file::AccessFlags::READWRITE));
-}
-
-TEST_F(FileClose, test_strong_withopen_attr)
-{
-
-  hdf5::property::FileAccessList fapl2;
-  fapl2.close_degree(hdf5::property::CloseDegree::STRONG);
-  EXPECT_EQ(fapl2.close_degree(), hdf5::property::CloseDegree::STRONG);
-  auto file = hdf5::file::open("testclose.h5",
-				hdf5::file::AccessFlags::READONLY, fapl2);
-  auto root_group = file.root();
-
-  auto attr = root_group.attributes[0];
-  root_group.close();
-
-  // with CloseDegree::STRONG it closes also attr
-  EXPECT_NO_THROW(file.close());
-
-
-  // everything is close so file can be reopen in a different mode, i.e. READWRITE
-  EXPECT_NO_THROW(hdf5::file::open("testclose.h5",
-				   hdf5::file::AccessFlags::READWRITE));
-}
-
-TEST_F(FileClose, test_strong_withopen_root)
-{
-
-  hdf5::property::FileAccessList fapl2;
-  fapl2.close_degree(hdf5::property::CloseDegree::STRONG);
-  EXPECT_EQ(fapl2.close_degree(), hdf5::property::CloseDegree::STRONG);
-  auto file = hdf5::file::open("testclose.h5",
-				hdf5::file::AccessFlags::READONLY, fapl2);
-  auto root_group = file.root();
-
-  // with CloseDegree::STRONG it closes also attr
-  file.close();
-
-  // everything is close so file can be reopen in a different mode, i.e. READWRITE
-  EXPECT_NO_THROW(hdf5::file::open("testclose.h5",
-				   hdf5::file::AccessFlags::READWRITE));
-}
-
-TEST_F(FileClose, test_withopen_attr_root)
-{
-
-  hdf5::property::FileAccessList fapl2;
-  EXPECT_EQ(fapl2.close_degree(), hdf5::property::CloseDegree::DEFAULT);
-  auto file = hdf5::file::open("testclose.h5",
-				hdf5::file::AccessFlags::READONLY, fapl2);
-  auto root_group = file.root();
-
-  auto attr = root_group.attributes[0];
-  EXPECT_EQ(file.count_open_objects(file::SearchFlags::ALL), 3ul);
-
-  // without CloseDegree::STRONG root_group and attr are still open
-  EXPECT_NO_THROW(file.close());
-
-
-  // file cannot be reopen in a different mode, i.e. READWRITE
-  EXPECT_THROW(hdf5::file::open("testclose.h5",
-				hdf5::file::AccessFlags::READWRITE),
-	       std::runtime_error);
-}
-
-TEST_F(FileClose, test_withopen_attr)
-{
-
-  hdf5::property::FileAccessList fapl2;
-  EXPECT_EQ(fapl2.close_degree(), hdf5::property::CloseDegree::DEFAULT);
-  auto file = hdf5::file::open("testclose.h5",
-				hdf5::file::AccessFlags::READONLY, fapl2);
-  auto root_group = file.root();
-
-  auto attr = root_group.attributes[0];
-  root_group.close();
-
-  EXPECT_EQ(file.count_open_objects(file::SearchFlags::ALL), 2ul);
-
-  // without CloseDegree::STRONG root_group and attr are still open
-  EXPECT_NO_THROW(file.close());
-
-
-  // file cannot be reopen in a different mode, i.e. READWRITE
-  EXPECT_THROW(hdf5::file::open("testclose.h5",
-				hdf5::file::AccessFlags::READWRITE),
-	       std::runtime_error);
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wexit-time-destructor"
```

Comment:
this should be `-Wexit-time-destructors` :)

Reply:
You are absolutely right about this.